biz.aQute.repository/src/aQute/bnd/repository/maven/pom/provider/BndPomRepository.java

```diff
@@ -75,43 +77,71 @@
 	private String				queryUrl;
 	private ScheduledFuture<?>	pomPoller;
 
-	public synchronized void init() {
-		try {
-			if (inited)
-				return;
-			inited = true;
-
-			Workspace workspace = registry.getPlugin(Workspace.class);
-			HttpClient client = registry.getPlugin(HttpClient.class);
-			localRepo = IO.getFile(configuration.local(MAVEN_REPO_LOCAL));
-			File location = workspace.getFile(getLocation());
-
-			List<MavenBackingRepository> release = MavenBackingRepository.create(configuration.releaseUrls(), reporter,
-				localRepo, client);
-			List<MavenBackingRepository> snapshot = MavenBackingRepository.create(configuration.snapshotUrls(),
-				reporter, localRepo, client);
-
-			MavenRepository repository = new MavenRepository(localRepo, name, release, snapshot, client.promiseFactory()
-				.executor(), reporter);
-
-			boolean transitive = configuration.transitive(true);
-
-			if (pomFiles != null) {
-				repoImpl = new PomRepository(repository, client, location, transitive).uris(pomFiles);
-			} else if (revisions != null) {
-				repoImpl = new PomRepository(repository, client, location, transitive).revisions(revisions);
-			} else if (query != null) {
-				repoImpl = new SearchRepository(repository, location, query, queryUrl, workspace, client, transitive);
-			} else {
-				repository.close();
-				throw new IllegalStateException("We have neither a pom, revision, or query set!");
-			}
-			bridge = new BridgeRepository(repoImpl);
+	private String				status;
 
-			startPoll();
-		} catch (Exception e) {
-			reporter.exception(e, "Init for BndPomRepository failed %s", configuration);
-			throw Exceptions.duck(e);
+	@SuppressWarnings("deprecation")
+	@Override
+	public void prepare() throws Exception {
+		synchronized (this) {
```

Comment:
prepare does not need to be synchronized, it is only called once after all plugins have been created and injected

Reply:
The old init method has been synchronized and the setProperties sets the the initialized marker in a syncronized block. I wasn't sure if this would be necessary, but I thought that their might be a reason, I could not see, so I decided to keep the sync approach. I will throw this out.